### PR TITLE
Include cross validation

### PR DESCRIPTION
**Pull request by wellgrisa**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -95,6 +95,40 @@ const schemaValidator = {
 }
 ```
 
+## Validating using cross properties (body, params or query)
+
+The middleware puts the req in the context. Therefore it's possible to use the other properties to validate. Remember that to make use of `this` it's necessary to use a function (not an arrow one).
```
Comment (wgrisa) [inline README.md]: You can cross-validate properties of your payload using a custom Yup test and accessing them by calling `this.options.context`.

File: README.md [hunk 2]
```diff
@@ -95,6 +95,40 @@ const schemaValidator = {
 }
 ```
 
+## Validating using cross properties (body, params or query)
```
Comment (wgrisa) [inline README.md]: Cross-validation using any properties from your payload

File: README.md [hunk 3]
```diff
@@ -95,6 +95,40 @@ const schemaValidator = {
...
+
+The middleware puts the req in the context. Therefore it's possible to use the other properties to validate. Remember that to make use of `this` it's necessary to use a function (not an arrow one).
+
+```ts
+const schemaValidator = {
+  schema: {
+    body: {
+      yupSchema: Yup.object().shape({
+        numberToValidate: Yup.string().test({
+          message: 'Check if your number correspond with the type given',
```
Comment (wgrisa) [inline README.md]: Check if your number corresponds with the given type

File: README.md [hunk 4]
```diff
@@ -95,6 +95,40 @@ const schemaValidator = {
...
+          test(this: Yup.TestContext, numberToValidate: any) {
+            const mod = numberToValidate % 2
+            // As you can see in the next line, the req is passed to the context
+            const type = this.options.context['req'].params.type
+
+            if (!type) {
+              return false
+            }
+
+            if (type === 'even') return mod === 0
```
Comment (wgrisa) [inline README.md]: Add the brackets for these if statements.

File: test/yup-middleware.spec.ts [hunk 5]
```diff
@@ -195,4 +198,125 @@ describe('express yup middleware', () => {
...
+            {
+              propertyPath: 'numberToValidate',
+              message: 'Check if your number correspond with the type given',
+            },
+          ],
+        },
+      })
+    })
+
+    it('returns a bad request error using a param value within a body validation', async () => {
```
Comment (wgrisa) [inline test/yup-middleware.spec.ts]: returns a bad request error when the number sent in the payload does not match the type param

File: test/yup-middleware.spec.ts [hunk 6]
```diff
@@ -195,4 +198,125 @@ describe('express yup middleware', () => {
...
+          numberToValidate: 2,
+        })
+        .expect(200)
+
+      expect(body).toStrictEqual({
+        errors: {
+          body: [
+            {
+              propertyPath: 'numberToValidate',
+              message: 'Check if your number correspond with the type given',
```
Comment (wgrisa) [inline test/yup-middleware.spec.ts]: Check if your number corresponds with the given type

File: test/yup-middleware.spec.ts [hunk 7]
```diff
@@ -195,4 +198,125 @@ describe('express yup middleware', () => {
...
+        .send({
+          numberToValidate: 2,
+        })
+        .expect(200)
+
+      expect(body).toStrictEqual({
+        errors: {
+          body: [
+            {
+              propertyPath: 'numberToValidate',
```
Comment (wgrisa) [inline test/yup-middleware.spec.ts:318]: typeMismatch

File: test/yup-middleware.spec.ts [hunk 8]
```diff
@@ -195,4 +198,125 @@ describe('express yup middleware', () => {
...
+      })
+    })
+
+    it('returns a bad request error using a param value within a body validation', async () => {
+      const schemaValidator: ExpressYupMiddlewareInterface = {
+        schema: {
+          body: {
+            yupSchema: Yup.object().shape({
+              numberToValidate: Yup.string().test({
+                message: 'Check if your number correspond with the type given',
```
Comment (wgrisa) [inline test/yup-middleware.spec.ts]: Check if your number corresponds with the given type

**Review by wgrisa — CHANGES_REQUESTED** [review]

File: test/yup-middleware.spec.ts (hunk 7 above)
Comment (wellgrisa) [reply test/yup-middleware.spec.ts:318]: Did not understand :(

File: src/schema-validator.ts [hunk 9]
```diff
@@ -17,7 +17,13 @@ export const validatePayload = async ({
     const propertySchema = schemaValidator.schema[propertyToValidate]
 
     try {
-      await propertySchema?.yupSchema.validate(payload[propertyToValidate], propertySchema.validateOptions)
+      await propertySchema?.yupSchema.validate(payload[propertyToValidate], {
+        ...propertySchema.validateOptions,
+        context: {
+          req: payload,
```
Comment (wgrisa) [inline src/schema-validator.ts]: use payload instead of req as it can be anything, not necessarily an express request object

**Review by wgrisa — CHANGES_REQUESTED** [review]

**Review by wgrisa — APPROVED** [review]

**Pull request merged**